### ISSUE TITLE
Incorrect Header capitalization

### ISSUE THREAD

**Issue by physics-sp** [kgretzky/evilginx2#499]
Hi,
Some headers are sent with an incorrect capitalization, for example:
The "Do not Track" header
instead of being sent like this: `DNT: 1`
is sent this way: `Dnt: 1`

In Google, instead of sending this header: `Google-Accounts-XSRF: 1`
this header is sent: `Google-Accounts-Xsrf: 1`

Thank you!

**Comment by kgretzky** [comment]
Thank you for reporting!

This may be an issue with the `go-proxy` library I'm using. Will keep an eye on that and may need to check their code.

**Comment by physics-sp** [comment]
Great! Sorry that I didn't provide a PR here, my go skills are very limited haha

**Comment by physics-sp** [comment]
Well, I have investigated a little bit.  

As you theorized, the issue is in the go proxy library.

[here](https://github.com/elazarl/goproxy/blob/master/proxy.go#L38-L49), you can see how they copy the headers:
```golang
func copyHeaders(dst, src http.Header, keepDestHeaders bool) {
	if !keepDestHeaders {
		for k := range dst {
			dst.Del(k)
		}
	}
	for k, vs := range src {
		for _, v := range vs {
			dst.Add(k, v)
		}
	}
}
```
They use `Add`. [here](https://golang.org/src/net/http/header.go), you can read it's implementation (inside `/net/http`).
```golang
// Add adds the key, value pair to the header.
// It appends to any existing values associated with key.
// The key is case insensitive; it is canonicalized by
// CanonicalHeaderKey.
func (h Header) Add(key, value string) {
	textproto.MIMEHeader(h).Add(key, value)
}
```
It says that the header is "canonicalized" via `CanonicalHeaderKey`.  

[here](https://golang.org/pkg/net/textproto/#CanonicalMIMEHeaderKey) clarifies what `CanonicalMIMEHeaderKey` actually does:
```
CanonicalMIMEHeaderKey returns the canonical format of the MIME header key s. The canonicalization converts the first  
letterand any letter following a hyphen to upper case; the rest are converted to lowercase. For example, the canonical key for 
 "accept-encoding" is "Accept-Encoding". MIME header keys are assumed to be ASCII only. If s contains a space or invalid  
header field bytes, it is returned without modifications. 
```

This also happens when you use the `Set` method.  

So, long story short, you can't do much about it. The documentation says:  `To use non-canonical keys, assign to the map directly.`  
Which is not really in your control, at least for headers like `DNT` and `Google-Accounts-XSRF`.  
Maybe a PR to the proxy repo to allow that option?  

Feel free to close this issue 😢

**Comment by physics-sp** [comment]
This can be "fixed" manually by modifying the source code and recompiling.
Just replace _Dnt_ with _DNT_ (or whatever header one might want to change) by modifying the `Header` map directly.

This should be done per phishing campaign but is possible.

**Comment by physics-sp** [comment]
In case anybody wonders how this can be done:
```golang
DNT := req.Header.Get("Dnt")
if DNT != "" {
	delete(req.Header, "Dnt");
	req.Header["DNT"] = []string{DNT}
}
```
